contracts/blender/src/contract.rs

```diff
@@ -160,6 +176,31 @@ mod blender {
             self.current_root()
         }
 
+        /// Retrieve the path from the leaf to the root. `None` if the leaf does not exist.
+        #[ink(message, selector = 4)]
+        pub fn merkle_path(&self, leaf_idx: u32) -> Option<MerklePath> {
+            if self.max_leaves > leaf_idx || leaf_idx >= self.next_free_leaf {
+                return None;
+            }
+
+            let mut current_idx = leaf_idx;
+            let mut path: MerklePath = vec![];
+            while current_idx > 1 {
+                let (left_sibling_idx, right_sibling_idx) = Self::get_sibling_indices(current_idx);
+                path.push(self.tree_value(left_sibling_idx));
+                path.push(self.tree_value(right_sibling_idx));
+                current_idx /= 2;
+            }
+
+            Some(path)
+        }
+
+        /// Check whether `nullifier` has been already used.
+        #[ink(message, selector = 5)]
+        pub fn observed_nullifier(&self, nullifier: Nullifier) -> bool {
```

Comment:
Personally, I would use `contains_*` instead since it's a standard for map-like types.

Reply:
changed